### DISCUSSION TITLE
Http Post Basic Auth credentials stored on environment variables

### DISCUSSION THREAD

**Discussion by tl-Bruno-Braga** [jertel/elastalert2#216] [Q&A] [answered]
Hello

Since some credentials can be passed through environment variables (e.g. ElasticSearch) or files (e.g. jira creds), would it be possible to use the same for basic authorization with http posts?

From a CI/CD perspective it would be more practical to read the creds from environment variables injected into the containers. This has been mentioned on other github issues but the solutions were hacky.

Thanks

**Comment by nsano-rururu** [comment]
@tl-Bruno-Braga 

I think that basic authentication is not supported by http_alerter, but is it actually moving?

**Comment by nsano-rururu** [comment, marked as answer]
@tl-Bruno-Braga

It is possible to use environment variables in the rule because the following pull requests are included.

https://github.com/jertel/elastalert2/pull/149

**Comment by jertel** [comment]
@nsano-rururu Basic auth is supported in http-post alerts via the header setting. The example in the docs shows one way to do this. https://elastalert2.readthedocs.io/en/latest/ruletypes.html#http-post

@tl-Bruno-Braga Could you tell me what prompted you to create an issue for this topic, instead of creating a discussion? I've tried to communicate that questions do not belong in this issues list (See default issue template, pinned discussion post, and project guidelines document), but the community continues to open issues instead of discussions. So if you could help me understand your reasoning for posting the question here that might help me improve this.

**Comment by nsano-rururu** [comment]
@jertel

The following "123dr3234" actually specifies the character string in which the user ID: password is Base64-encoded.

```
http_post_headers:
  authorization: Basic 123dr3234
```

**Comment by tl-Bruno-Braga** [comment]
Thanks for reaching out fellas. 

Apologies for creating the issue instead of discussion @jertel .

**Reply by jertel** [reply]
No problem! Always looking for ways to improve things. Converted it to a discussion so others can benefit from searches.
